src/data_structure/mga_data.jl

```diff
@@ -0,0 +1,219 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+function units_invested_mga_indices()
+    unique(
+        [
+        (unit=ug,)
+        for ug in unit(units_invested_mga=true)])
+end
+
+function units_invested_mga_indices(mga_iteration)
+    unique(
+        [
+        (unit=ug, mga_iteration=mga_it)
+        for ug in unit(units_invested_mga=true)
+            for mga_it in mga_iteration])
+end
+
+function connections_invested_mga_indices()
+    unique(
+        [
+        (connection=cg,)
+        for cg in connection(connections_invested_mga=true)])
+end
+
+function connections_invested_mga_indices(mga_iteration)
+    unique(
+        [
+        (connection=cg, mga_iteration=mga_it)
+        for cg in connection(connections_invested_mga=true)
+            for mga_it in mga_iteration])
+end
+
+function storages_invested_mga_indices()
+    unique(
+        [
+        (node=ng, )
+        for ng in node(storages_invested_mga=true)])
+end
+
+function storages_invested_mga_indices(mga_iteration)
+    unique(
+        [
+        (node=ng, mga_iteration=mga_it)
+        for ng in node(storages_invested_mga=true)
+            for mga_it in mga_iteration])
+end
+
+function set_objective_mga_iteration!(m;iteration=nothing)
+    instance = m.ext[:instance]
+    if !mga_diff_relative(model=instance) #FIXME: define this properly for relative and not relative
+        _set_objective_mga_iteration!(
+            m,
+            :units_invested_available,
+            units_invested_available_indices,
+            unit_stochastic_scenario_weight,
+            units_invested_mga_indices,
+            units_invested_big_m_mga,
+            iteration
+        )
+        _set_objective_mga_iteration!(
+            m,
+            :connections_invested_available,
+            connections_invested_available_indices,
+            connection_stochastic_scenario_weight,
+            connections_invested_mga_indices,
+            connections_invested_big_m_mga,
+            iteration
+        )
+        _set_objective_mga_iteration!(
+            m,
+            :storages_invested_available,
+            storages_invested_available_indices,
+            node_stochastic_scenario_weight,
+            storages_invested_mga_indices,
+            storages_invested_big_m_mga,
+            iteration
+        )
+        @fetch mga_aux_diff, mga_objective = m.ext[:variables]
+        ub_objective = get!(m.ext[:constraints],:mga_objective_ub,Dict())
+        ub_objective[iteration] = @constraint(
+                m,
+                mga_objective[(model = m.ext[:instance],t=current_window(m))]
+                <= sum(
+                mga_aux_diff[ind...]
+                       for ind in vcat([storages_invested_mga_indices(iteration),connections_invested_mga_indices(iteration),units_invested_mga_indices(iteration)])

```

Comment:
please indent the `for` at the same level of the previous line as is our style for comprehensions.